src/main/java/com/aws/greengrass/tes/HttpServerImpl.java

```diff
@@ -31,25 +28,32 @@ public class HttpServerImpl implements Server {
      * @param executorService executor service instance
      * @throws IOException When server creation fails
      */
-    HttpServerImpl(int port, HttpHandler credentialRequestHandler, ExecutorService executorService) throws IOException {
-        httpImpl = HttpServer.create(new InetSocketAddress("localhost", port), 0);
+    HttpServerImpl(int port, HttpHandler credentialRequestHandler, ExecutorService executorService)
+            throws IOException {
+        this.configuredPort = port;
         this.credentialRequestHandler = credentialRequestHandler;
         this.executorService = executorService;
     }
 
     @Override
-    public void start() {
-        httpImpl.createContext(URL, credentialRequestHandler);
-        httpImpl.setExecutor(executorService);
-        httpImpl.start();
+    public void start() throws IOException {
+        server = HttpServer.create(new InetSocketAddress(ANY_INTERFACE, configuredPort), 0);
```

Comment:
so this makes us listen on any interface for both ipv4 and ipv6?

Doesn't this also mean that TES is available on the LAN (we don't want that)?

Reply:
Why are we starting a new server within `start()`, that makes start non-idempotent. We could end up leaking server instances it seems.

Reply:
> so this makes us listen on any interface for both ipv4 and ipv6?
> 
> Doesn't this also mean that TES is available on the LAN (we don't want that)?

Yes and yes.  So even though we check Authorization header, reason is we don't want to increase attack surface?  
Will revise

> Why are we starting a new server within `start()`, that makes start non-idempotent. We could end up leaking server instances it seems.

No reason, was just removing network binding from the HttpServerImpl constructor, will revert